dataloader.py

```diff
@@ -225,14 +230,18 @@ def build_dataloader(configuration):
     )
     dataset_size = len(dataset)
     indices = list(range(dataset_size))
-    split = int(np.floor(validation_split * dataset_size))
+    split = int(np.floor(separate_split * dataset_size))
 
     np.random.seed(configuration.seed)
     np.random.shuffle(indices)
-    train_indices, val_indices = indices[split:], indices[:split]
+    train_indices, val_indices, test_indices = (
+        indices[split:],
+        indices[:split],
+        indices[split:],
+    )
```

Comment:
The same indices are used here for the training and test sets, which makes them identical. We want to avoid this, it's important that the test set is completely distinct and separate from the training and validation sets.